pkg/networkpolicy/adminnetworkpolicy.go

```diff
@@ -0,0 +1,239 @@
+			// If any AdminNetworkPolicyEgressPeer matches the destination of outgoing
+			// traffic then the specified action is applied.
+			// This field must be defined and contain at least one item.
+			for _, to := range rule.To {
+				// Exactly one of the selector pointers must be set for a given peer. If a
+				// consumer observes none of its fields are set, they must assume an unknown
+				// option has been specified and fail closed.
+				if to.Namespaces != nil {
+					if c.namespaceSelector(to.Namespaces, dstPod) {
+						return rule.Action
+					}
+				}
+
+				if to.Pods != nil {
+					if c.namespaceSelector(&to.Pods.NamespaceSelector, dstPod) &&
+						podSelector(&to.Pods.PodSelector, dstPod) {
+						return rule.Action
+					}
+				}
+
+				if to.Nodes != nil {
+					// TODO
+					// Needs to watch nodes objects
+				}
+
+				for _, network := range to.Networks {
+					_, cidr, err := net.ParseCIDR(string(network))
+					if err != nil { // this has been validated by the API
+						continue
+					}
+					if cidr.Contains(dstIP) {
+						return rule.Action
+					}
+				}
+			}
+		}
+	}
+
+	// 2. dstPod ingress is accepted
+	for _, policy := range dstPodAdminNetworkPolices {
+		// Ingress is the list of Ingress rules to be applied to the selected pods. A total of 100 rules will be allowed in each ANP instance. The relative precedence of ingress rules within a single ANP object (all of which share the priority) will be determined by the order in which the rule is written. Thus, a rule that appears at the top of the ingress rules would take the highest precedence.
+		// ANPs with no ingress rules do not affect ingress traffic.
+		for _, rule := range policy.Spec.Ingress {
+			// Ports allows for matching traffic based on port and protocols.
+			// This field is a list of destination ports for the outgoing egress traffic.
+			// If Ports is not set then the rule does not filter traffic via port.
+			if rule.Ports != nil {
+				if !c.evaluateAdminNetworkPolicyPort(*rule.Ports, srcPod, srcPort, protocol) {
+					continue
+				}
+			}
+			// To is the List of destinations whose traffic this rule applies to.
+			// If any AdminNetworkPolicyEgressPeer matches the destination of outgoing
+			// traffic then the specified action is applied.
+			// This field must be defined and contain at least one item.
+			for _, from := range rule.From {
+				// Exactly one of the selector pointers must be set for a given peer. If a
+				// consumer observes none of its fields are set, they must assume an unknown
+				// option has been specified and fail closed.
+				if from.Namespaces != nil {
+					if c.namespaceSelector(from.Namespaces, srcPod) {
+						return rule.Action
+					}
+				}
+
+				if from.Pods != nil {
+					if c.namespaceSelector(&from.Pods.NamespaceSelector, srcPod) &&
+						podSelector(&from.Pods.PodSelector, srcPod) {
+						return rule.Action
+					}
+				}
+			}
+
+		}
+	}
+
+	return npav1alpha1.AdminNetworkPolicyRuleActionPass
```

Comment:
ah so no ANPs match for the pod is considered as a pass so that netpol's can be evaluated?

Reply:
exactly https://github.com/kubernetes-sigs/kube-network-policies/pull/9#discussion_r1575377948